pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -197,16 +223,20 @@ func (w *Workflow) GetExecSpec() *WorkflowSpec {
 }
 
 func (w *Workflow) HasArtifactGC() bool {
+
+	if w.Spec.ArtifactGC != nil && w.Spec.ArtifactGC.Strategy != ArtifactGCNever {
```

Comment:
I think you could add `spec.GetArifactGC().Strategy` with nil check in `GetArifactGC()` that return empty struct